### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by CarlosM10011**
This pull requests completes the actions in the week 2 walkthrough. Changes include edits to the website landing page and a dedicated page for generating random facts.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -13,16 +13,18 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomFact() {
```
Comment (hzimz) [inline portfolio/src/main/webapp/script.js]: Here and elsewhere, prefer to use arrow functions. https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions

**Review by hzimz — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
-  </body>
+</head>
+
+<body>
+    <nav id="top-nav-bar">
+        <a href="/">Home</a>
+        <a href="/random.html">random generator</a>
+    </nav>
+    <main id="content">
+        <H1>Carlos</H1>
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/index.html]: HTML tags should be lowercase.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+        <p>My name is Carlos, and I am currently a rising junior Computer Science student at Tennessee Tech University. When not breaking things code wise, I am doing something like reading or listening to music.</p>
+        <p>I also enjoy pondering the meaning of life. Specifically, considering
+            why it is the number 42. Refer to
+            <a href="https://en.wikipedia.org/wiki/42_(number)#The_Hitchhiker's_Guide_to_the_Galaxy">
+                this Wikipedia article</a>, which explains it.</p>
+        <p>I normally do not force Windows to crash on a daily basis, but here
+            is a screenshot captured by VMware Workstation.</p>
+        <img src="/images/crash.png" alt="Windows 10 frozen on a BSOD.">
+    </main>
+    <a href="/#top-nav-bar">Back to top</a>
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/index.html]: 80 character lines.

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
-      <div id="greeting-container"></div>
-    </div>
-  </body>
+</head>
+
+<body>
+    <nav id="top-nav-bar">
+        <a href="/">Home</a>
+        <a href="/random.html">random generator</a>
+    </nav>
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/index.html]: See if you can find a way to break out this navbar into it's own file so you don't have to copy and paste it in each file. This way, you only have to update it once and encourage code reuse.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -13,16 +13,18 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/script.js]: nit: one line for short comments like this.

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -1,18 +1,31 @@
 <!DOCTYPE html>
 <html>
-  <head>
+
+<head>
     <meta charset="UTF-8">
-    <title>My Portfolio</title>
+    <title>My Portfolio UNDER CONSTRUCTION</title>
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/index.html]: nit: format a little nicer. 

```
My Portfolio - Under Construction
```

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by hzimz — APPROVED** [review]

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (CarlosM10011) [reply portfolio/src/main/webapp/index.html]: Hmm, I may need to use a server side include for this. It looks like the server is statically serving the files.

Comment (CarlosM10011) [reply portfolio/src/main/webapp/index.html]: So I found a way to do server side includes using jsp; Week 3 introduces us to java servlets, and I discovered jsp by accident when looking up ways of embedding   java in html pages. I can always revert this latest commit and do it client side using js if there is a better way of doing this.

Comment (CarlosM10011) [reply portfolio/src/main/webapp/index.html]: I reverted and forked the last commit; I moved the nav bar to a js file.

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**